### PR TITLE
DATAUP-497: more job message edits

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

Continue simplification of job-related messages.

jobCommChannel.js: 
- more tokenising of message strings
- remove unused `RESULT` and `PING` message types and tests for those messages
- pass most backend messages as-is to the frontend without any faffing around with "translating" them
- remove nesting / "options" in job logs requests

jobManager, jobStatusTable, jobInputParams, jobLogViewer, kbaseNarrativeJobStatus:
- adapt code to account for new message structure
- update tests to take into account all the above changes

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/DATAUP-497
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [x] Don't check the narrative yet as there are some more edits to go in first.

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: docs/design/job_architecture.md [hunk 1]
```diff
@@ -48,10 +48,9 @@ All the `request-job-*` requests take as arguments either a single job ID string
 `request-job-log` - request the job logs starting at some given line.
   * `jobId` - a string, the job id OR
   * `jobIdList` - an array of job IDs
-  * `options` - an object, with attributes:
-    * `first_line` - the first line (0-indexed) to request
-    * `num_lines` - the number of lines to request (will get back up to that many if there aren't more)
-    * `latest` -  true if requesting just the latest set of logs
+  * `first_line` - the first line (0-indexed) to request (optional)
+  * `num_lines` - the number of lines to request (will get back up to that many if there aren't more) (optional)
+  * `latest` -  true if requesting just the latest set of logs (optional)
```
Comment (ialarmedalien) [inline docs/design/job_architecture.md:53]: flatten out these params

File: kbase-extension/static/kbase/js/common/jobCommChannel.js [hunk 2]
```diff
@@ -395,14 +400,11 @@ define([
...
-                // Object with keys jobId and values { jobInfo: { ...job params... } }
                 case BACKEND_RESPONSES.INFO:
                     Object.keys(msgData).forEach((_jobId) => {
                         const jobData = msgData[_jobId];
-                        if (Jobs.isValidJobInfoObject(jobData)) {
-                            this.sendBusMessage(JOB, _jobId, RESPONSES.INFO, {
-                                jobInfo: jobData,
-                            });
+                        if (this.validationFn[msgType](jobData)) {
+                            this.sendBusMessage(JOB, _jobId, RESPONSES.INFO, jobData);
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobCommChannel.js:407]: job info sent as-is, no longer wrapped in `{ jobInfo: ... }`

File: kbase-extension/static/kbase/js/common/jobCommChannel.js [hunk 3]
```diff
@@ -412,11 +414,8 @@ define([
                 case BACKEND_RESPONSES.LOGS:
                     Object.keys(msgData).forEach((_jobId) => {
                         const jobData = msgData[_jobId];
-                        if (Jobs.isValidJobLogsObject(jobData)) {
-                            this.sendBusMessage(JOB, _jobId, RESPONSES.LOGS, {
-                                jobId: _jobId,
-                                logs: jobData,
-                            });
+                        if (this.validationFn[msgType](jobData)) {
+                            this.sendBusMessage(JOB, _jobId, RESPONSES.LOGS, jobData);
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobCommChannel.js:418]: Ditto for the `job-logs` data

File: kbase-extension/static/kbase/js/util/jobLogViewer.js [hunk 4]
```diff
@@ -1207,26 +1205,25 @@ define([
...
                  */
                 this.state.awaitingLog = false;
 
                 if (message.error) {
                     return this.handleLogDeleted(message);
                 }
-                if (message.logs.lines.length !== 0) {
-                    const viewLines = message.logs.lines.map((line) => {
+                if (message.lines.length !== 0) {
+                    const viewLines = message.lines.map((line) => {
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/util/jobLogViewer.js:1216]: remove the nesting -- the data is no longer in the `logs` object.

File: kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeJobStatus.js [hunk 5]
```diff
@@ -184,10 +180,10 @@ define([
         },
 
         handleJobInfo: function (info) {
-            if (utils.getCellMeta(this.cell, 'kbase.attributes.title') !== info.jobInfo.app_name) {
+            if (utils.getCellMeta(this.cell, 'kbase.attributes.title') !== info.app_name) {
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeJobStatus.js:183]: remove the `jobInfo` wrapper

File: test/unit/spec/common/cellComponents/tabs/jobStatusTable-Spec.js [hunk 6]
```diff
@@ -398,14 +398,7 @@ define([
      */
     function updateInfo(ctx) {
         const { jobId, jobInfo } = ctx;
-        sendBusMessage(
-            ctx,
-            {
-                jobInfo,
-            },
```
Comment (ialarmedalien) [inline test/unit/spec/common/cellComponents/tabs/jobStatusTable-Spec.js:405]: remove `jobInfo` wrapper

File: test/unit/spec/common/jobCommChannel-spec.js [hunk 7]
```diff
@@ -519,23 +512,6 @@ define([
                     },
                 ],
             },
-            {
```
Comment (ialarmedalien) [inline test/unit/spec/common/jobCommChannel-spec.js:522]: unused

File: test/unit/spec/util/jobLogViewerSpec.js [hunk 8]
```diff
@@ -66,11 +66,10 @@ define([
...
         return formatMessage(jobId, 'logs', {
-            logs: {
-                first: 0,
-                max_lines: logMessages.length,
-                lines: logMessages,
-            },
+            job_id: jobId,
+            first: 0,
+            max_lines: logMessages.length,
+            lines: logMessages,
```
Comment (ialarmedalien) [inline test/unit/spec/util/jobLogViewerSpec.js:72]: no `logs` wrapper

**Review by briehl — APPROVED** [review]
🎉

**Pull request merged**